futures/tests/all.rs

```diff
@@ -268,6 +268,14 @@ fn join_all_iter_lifetime() {
     assert_done(|| sizes(vec![&[1,2,3], &[], &[0]]), Ok(vec![3, 0, 1]));
 }
 
+#[test]
+fn join_all_from_iter() {
+    assert_done(
+        || vec![f_ok(1), f_ok(2)].into_iter().collect::<JoinAll<_>>(),
```

Comment:
Is the turbofish actually necessary in general, or can it be inferred?

Reply:
It is unfortunately necessary in general: at the call site, you know that you need some T: Future, but you don’t know whit T specifically are you interested in.